src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -48,39 +54,47 @@ final case class ThermalGrid(
 
   /** Determine the energy demand of the total grid at the given instance in
     * time and returns it including the updatedState.
-    *
-    * @param lastHpState
+    * @param tick
+    *   The actual tick of simulation.
+    * @param state
     *   Last state of the heat pump.
-    * @param relevantData
-    *   Required data for calculation.
+    * @param operatingPoint
+    *   The operating point of the heat pump.
     * @return
     *   The total energy demand of the house and the storage and an updated
     *   [[ThermalGridState]].
     */
   def energyDemandAndUpdatedState(
-      relevantData: HpRelevantData,
-      lastHpState: HpState,
+      tick: Long,
+      state: HpState,
+      operatingPoint: HpOperatingPoint,
   ): (ThermalDemandWrapper, ThermalGridState) = {
```

Comment:
I got confused looking through the code and was trying to pinpoint the source of my confusion, and I think it's this.

I think energy demand and current state calculation should be separated. For calculating the current state, you should only need the last state and operating point since then. Energy demand on the other hand is something that should lead to the determiantion of the operating point (or flex options), and should take the current state only as input (plus current weather data, but that's besides the point).